Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift

```diff
@@ -253,11 +272,61 @@ extension PostCampaignCheckoutViewController: PledgeViewCTAContainerViewDelegate
 
   func submitButtonTapped() {
     self.paymentMethodsViewController.cancelModalPresentation(true)
-    // TODO: Respond to button tap
+    self.viewModel.inputs.submitButtonTapped()
   }
 
   func termsOfUseTapped(with helpType: HelpType) {
     self.paymentMethodsViewController.cancelModalPresentation(true)
     self.viewModel.inputs.termsOfUseTapped(with: helpType)
   }
 }
+
+// MARK: - PledgePaymentMethodsViewControllerDelegate
+
+extension PostCampaignCheckoutViewController: PledgePaymentMethodsViewControllerDelegate {
+  func pledgePaymentMethodsViewController(
+    _: PledgePaymentMethodsViewController,
+    didSelectCreditCard paymentSource: PaymentSourceSelected
+  ) {
+    switch paymentSource {
+    case let .paymentIntentClientSecret(clientSecret):
+      return STPAPIClient.shared.retrievePaymentIntent(withClientSecret: clientSecret) { intent, _ in
```

Comment:
It would follow our existing patterns better if this call to Stripe's API went into `PostCampaignPledgeViewModel`, instead - that way, all the business logic stays in the VM, including the call back out to the error banner.

Reply:
🤔 Unless we explicitly keep stripe stuff out of VMs so that the VMs are more testable. Could go either way on this one, now that I say it.

Reply:
Sounds good, OK.